src/Kiota.Builder/OpenApiExtensions/OpenApiLogoExtension.cs

```diff
@@ -1,20 +1,41 @@
-﻿using Microsoft.OpenApi.Any;
+﻿using System;
+using Kiota.Builder.Extensions;
+using Microsoft.OpenApi;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Writers;
 
 namespace Kiota.Builder.OpenApiExtensions;
 
-public class OpenApiLogoExtension : OpenApiSimpleStringExtension
+public class OpenApiLogoExtension : IOpenApiExtension
 {
     public static string Name => "x-logo";
-    public string? Logo
+#pragma warning disable CA1056
+    public string? Url
+#pragma warning restore CA1056
     {
         get; set;
     }
-    protected override string? ValueSelector => Logo;
     public static OpenApiLogoExtension Parse(IOpenApiAny source)
     {
-        return new OpenApiLogoExtension
+        if (source is not OpenApiObject rawObject) throw new ArgumentOutOfRangeException(nameof(source));
```

Comment:
In this case, should the method accept an `OpenApiObject` instead?

Reply:
The contract from the openApi library has the input as a `IOpenApiAny`. So we would need to do the validation/checking either way. https://github.com/microsoft/kiota/blob/feature/plugins/src/Kiota.Builder/OpenApiDocumentDownloadService.cs#L116

Reply:
Makes sense. Seen the pattern in the other extension classes. Thanks.